frontend/src/scenes/authentication/RegionSelect.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { useValues, path, kea, actions, useActions, listeners } from 'kea'
+import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
+import { LemonSelect } from '@posthog/lemon-ui'
+import { Region } from '~/types'
+import { CLOUD_EU_URL, CLOUD_US_URL } from 'lib/constants'
+import { router } from 'kea-router'
+
+import type { regionSelectLogicType } from './RegionSelectType'
+import { PureField } from 'lib/forms/Field'
+
+export const regionSelectLogic = kea<regionSelectLogicType>([
```

Comment:
I'm all for logics but honestly at this level of complexity I think the `onRegionChange` handler could just be inlined in `RegionSelect`

Reply:
I was thinking on expanding the logic, but then I didn't 😅  I removed it now!